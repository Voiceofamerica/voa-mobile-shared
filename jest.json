{
  "setupTestFrameworkScriptFile": "<rootDir>/.jest/setupTests.js",
  "transform": {
    "\\.(ts|tsx)$": "ts-jest/preprocessor.js",
    "\\.(js|jsx)$": "babel-jest"
  },
  "setupFiles": [
    "<rootDir>/.jest/rAF.js"
  ],
  "testRegex": "\\.spec\\.(ts|tsx)$",
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js",
    "json"
  ],
  "cacheDirectory": "jestCache",
  "globals": {
    "__DEV__": true,
    "ts-jest": {
      "tsConfigFile": "tsconfig.test.json"
    }
  },
  "moduleNameMapper": {
    "^components/(.*)$": "<rootDir>/components/$1/index.ts",
    "^helpers/(.*)$": "<rootDir>/helpers/$1.ts",
    "^types/(.*)$": "<rootDir>/styles/$1.d.ts",
    "\\.s?css$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|svg)$": "<rootDir>/__mocks__/fileMock.js"
  },
  "collectCoverage": true,
  "collectCoverageFrom": [
    "**/*.{ts,tsx}",
    "!**/index.{ts,tsx}",
    "!**/*.story.tsx",
    "!**/*.d.ts"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 45,
      "functions": 40,
      "lines": 50,
      "statements": 50
    }
  }
}