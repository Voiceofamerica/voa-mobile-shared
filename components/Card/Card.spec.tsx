
import 'jest'
import 'react-native'
import * as React from 'react'
import * as RX from 'reactxp'
import * as renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import OriginalCard, { ArticleBlurb } from './Card'

const importModule = (): typeof OriginalCard => {
  const module = require('.').default
  return module
}

const testArticle: ArticleBlurb = {
  image: 'https://gdb.voanews.com/3E1C440D-1C78-4B88-84F7-ECB12FB5B874_cx3_cy0_cw97_w650_r1.png',
  minor: '15:08',
  title: 'Pressident Trump: From North Korea to Iran, we confront dangerous governments',
}

const mockedComponents = [
]

const defaultMocks = [
]

const allMocks = mockedComponents.concat(defaultMocks)

describe('<Card />', () => {
  beforeEach(() => {
    jest.resetModules()
    mockedComponents.forEach(component => {
      jest.setMock(component, component)
    })
    defaultMocks.forEach(component => {
      jest.setMock(component, jest.genMockFromModule(component))
    })
  })

  afterEach(() => {
    allMocks.forEach(component => {
      jest.unmock(component)
    })
  })

  describe('snapshots', () => {
    it('renders correctly', () => {
      const Card = importModule()
      const tree = renderer.create(
        <Card onPress={jest.fn()} blurb={testArticle} />,
      )

      expect(tree).toMatchSnapshot()
    })
  })

  describe('button press', () => {
    it('should call onPress()', () => {
      const pressHandler = jest.fn()
      jest.unmock('reactxp')
      const RealRX = require('reactxp')

      const Card = importModule()
      const wrapper = shallow(
        <Card onPress={pressHandler} blurb={testArticle} />,
      )

      wrapper
        .find<any>(RealRX.View)
        .filterWhere(v => typeof v.props().onPress === 'function')
        .simulate('press')

      expect(pressHandler).toHaveBeenCalledTimes(1)
    })
  })
})
