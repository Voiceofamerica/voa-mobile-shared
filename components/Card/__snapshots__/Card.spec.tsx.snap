// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshots renders correctly 1`] = `
<RX.View
  onLayout={[Function]}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#DDD",
        "borderColor": "rgba(128, 128, 128, 0.5)",
        "borderRadius": 5,
        "borderWidth": 1,
        "elevation": 6,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
        "margin": 5,
        "overflow": "hidden",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 10,
        "shadowRadius": 10,
      },
      Object {
        "height": -7.5,
      },
    ]
  }
>
  <RX.Image
    resizeMethod="scale"
    resizeMode="cover"
    source="https://gdb.voanews.com/3E1C440D-1C78-4B88-84F7-ECB12FB5B874_cx3_cy0_cw97_w650_r1.png"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <RX.View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "transparent",
          "rgba(0, 0, 0, 0.8)",
        ]
      }
      style={
        Object {
          "flex": 1,
          "padding": 8,
          "paddingTop": 30,
        }
      }
    >
      <RX.Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#FFF",
              "paddingBottom": 5,
            },
            Object {
              "fontSize": NaN,
            },
          ]
        }
      >
        15:08
      </RX.Text>
      <RX.Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#FFF",
              "fontWeight": "bold",
            },
            Object {
              "fontSize": NaN,
            },
          ]
        }
      >
        Pressident Trump: From North Korea to Iran, we confront dangerous governments
      </RX.Text>
    </LinearGradient>
  </RX.View>
</RX.View>
`;
